devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -0,0 +1,129 @@
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (name: "ZigBee Battery Accessory Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.remotecontroller") {
+		capability "Actuator"
+		capability "Switch"
+		capability "Battery"
+		capability "Switch Level"
+		capability "Configuration"
+		
+		fingerprint profileId: "0104", inClusters: "0000,0001,0003,0020,FC11", outClusters: "0003,0004,0006,0008,FC10", manufacturer: "sengled", model: "E1E-G7F", deviceJoinName: "Sengled Smart Switch"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.battery", key: "SECONDARY_CONTROL") {
+				attributeState "battery", label: 'battery ${currentValue}%', unit: "%"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getDOUBLE_STEP() {10}
+def getSTEP() {5}
+
+def parse(String description) {
+
+	def descMap = zigbee.parseDescriptionAsMap(description)
+
+	if (descMap && descMap.clusterInt == 0xFC10) {
+		def currentLevel = device.currentValue("level") as Integer ?: 0
+		def value = 0
+		if (descMap.data[0] == '01') {
+			//short press of 'ON' button
+			sendEvent(name: "switch", value: "on")
+		} else if (descMap.data[0] == '02' ) {
+			// move up
+			if (descMap.data[2] == '02') {
+				//long press of 'BRIGHTEN' button
+				value = Math.min(currentLevel + DOUBLE_STEP, 100)
+			} else if (descMap.data[2] == '01') {
+				//short press of 'BRIGHTEN' button
+				value = Math.min(currentLevel + STEP, 100)
+			}
+			sendEvent(name: "switch", value: "on")
+			sendEvent(name: "level", value: value)
+			
+		} else if (descMap.data[0] == "03") {
+			//move down
+			if (descMap.data[2] == '02') {
+				//long press of 'DIM' button
+				value = Math.max(currentLevel - DOUBLE_STEP, 0)
+			} else if (descMap.data[2] == '01') {
+				//short press of 'DIM' button
+				value = Math.max(currentLevel - STEP, 0)
+			}
+			
+			if (value == 0) {
+				sendEvent(name: "switch", value: "off")
+				sendEvent(name: "level", value: value)
```

Comment:
this `sendEvent` doesn't need to be in the if statement or duplicated below.

Reply:
right, right, thanks